### PR TITLE
Add support for test type NUnit

### PR DESCRIPTION
**Pull request by nabobalis**
https://github.com/pytest-dev/pytest-nunit

Do we want to set a flag to toggle between the junit and nunit style?

This change does mean that the pytest package will need to be added to a package's setup.cfg or tox.ini file.

### CODE REVIEW

File: run-tox-env.yml [hunk 1]
```diff
@@ -170,12 +170,12 @@ jobs:
...
             testResultsFiles: '**/test-*.xml'
             testRunTitle: 'Publish test results for ${{ variables.friendly_name }}'
 
-        - ${{ if eq(coalesce(env['coverage'], parameters.coverage), 'azure') }}:
-          - task: PublishCodeCoverageResults@1
-            condition: succeededOrFailed()
-            inputs:
-              codeCoverageTool: Cobertura
-              summaryFileLocation: '$(System.DefaultWorkingDirectory)/coverage.xml'
+        - task: PublishCodeCoverageResults@1
```
Comment (nabobalis) [inline run-tox-env.yml:177]: Why not upload the code cov to azure anyway?

File: tox.ini [hunk 2]
```diff
@@ -18,9 +18,11 @@ skipsdist = true
 setenv = ENVNAME={envname}
 deps =
     pytest
+    pytest-cov
+    pytest-nunit
     opengl: vispy
     opengl: PyQt5
-commands = pytest test.py
+commands = pytest test.py {posargs}
```
Comment (nabobalis) [inline tox.ini:25]: Forgot this so we were not testing any extra args in the templates.

**Comment by nabobalis** [comment]
Codecov to azure is not working for the linux 32 bit build and I am not sure why.

**Comment by astrofrog** [comment]
Just to be clear, did you mean that pytest-nunit has to be added as a test dependency for any package using the template? If so I think this behaviour should probably be opt-in.

What are the benefits of switching from junit to nunit?

**Comment by nabobalis** [comment]
They promise better support for azure's test reporting compared to the default junit style. 

I had hoped to figure out to get pytest to register an external package outside of tox but I failed. So if people wanted to use this they would have to add the pytest package as an extra dependency.

I can setup a nunit flag instead?

File: run-tox-env.yml [hunk 3]
```diff
@@ -62,7 +69,7 @@ jobs:
           fetchDepth: 999999999
 
         - ${{ if eq(env_pair.key, 'linux32') }}:
-          - bash: docker run -v $PWD:/project -i --name manylinux2010 -d quay.io/pypa/manylinux2010_i686:latest /bin/bash
```
Comment (Cadair) [inline run-tox-env.yml:65]: I think the manylinux should be configurable. Some packages aren't built for newer ones and then you have to compile them from source which leads to long build times and issues we see on SunPy's builds. (bumping the versions further probably makes this even worse)

File: run-tox-env.yml [hunk 4]
```diff
@@ -170,12 +177,12 @@ jobs:
         - task: PublishTestResults@2
           condition: succeededOrFailed()
           inputs:
+            testResultsFormat: NUnit
```
Comment (Cadair) [inline run-tox-env.yml:180]: should this be dependant on the flag?

File: docs/run_tox_env.rst [hunk 5]
```diff
@@ -58,6 +58,25 @@ name of the service to use:
...
+
+    - template: run-tox-env.yml@OpenAstronomy
+      parameters:
+        coverage: codecov
+        envs:
+        - ...
+
+At this time, only ``codecov`` is supported.
+
+
```
Comment (Cadair) [inline docs/run_tox_env.rst:79]: needs some nunit docs?

Comment (nabobalis) [reply docs/run_tox_env.rst:79]: Sigh.

File: run-tox-env.yml (hunk 4 above)
Comment (nabobalis) [reply run-tox-env.yml:180]: Sigh.

File: run-tox-env.yml (hunk 3 above)
Comment (nabobalis) [reply run-tox-env.yml:65]: Sigh.

**Comment by nabobalis** [comment]
I stopped caring, someone else can finish this if they want.

**Pull request closed without merging**